core/chains/cosmos.go

```diff
@@ -18,70 +18,131 @@ import (
 
 const BONDED = "BOND_STATUS_BONDED"
 
+func Cosmos() (int, error) {
+	validatorURL := "https://proxy.atomscan.com/cosmoshub-lcd/cosmos/staking/v1beta1/validators?page.offset=1&pagination.limit=500&status=BOND_STATUS_BONDED"
+	poolURL := "https://proxy.atomscan.com/cosmoshub-lcd/cosmos/staking/v1beta1/pool"
+
+	return FetchCosmosSDKNakaCoeff("cosmos", validatorURL, poolURL)
+}
+
 type cosmosResponse struct {
 	Validators []struct {
-		Status string `json:"status"`
-		Tokens string `json:"tokens"`
+		OperatorAddress string `json:"operator_address"`
+		ConsensusPubkey struct {
+			Type string `json:"@type"`
+			Key  string `json:"key"`
+		} `json:"consensus_pubkey"`
+		Jailed          bool   `json:"jailed"`
+		Status          string `json:"status"`
+		Tokens          string `json:"tokens"`
+		DelegatorShares string `json:"delegator_shares"`
+		Description     struct {
+			Moniker         string `json:"moniker"`
+			Identity        string `json:"identity"`
+			Website         string `json:"website"`
+			SecurityContact string `json:"security_contact"`
+			Details         string `json:"details"`
+		} `json:"description"`
+		UnbondingHeight string `json:"unbonding_height"`
+		UnbondingTime   string `json:"unbonding_time"`
+		Commission      struct {
+			CommissionRates struct {
+				Rate          string `json:"rate"`
+				MaxRate       string `json:"max_rate"`
+				MaxChangeRate string `json:"max_change_rate"`
+			} `json:"commission_rates"`
+			UpdateTime string `json:"update_time"`
+		} `json:"commission"`
+		MinSelfDelegation       string   `json:"min_self_delegation"`
+		UnbondingOnHoldRefCount string   `json:"unbonding_on_hold_ref_count"`
+		UnbondingIds            []string `json:"unbonding_ids"`
+		ValidatorBondShares     string   `json:"validator_bond_shares"`
+		LiquidShares            string   `json:"liquid_shares"`
 	} `json:"validators"`
 }
 
-func Cosmos() (int, error) {
-	url := "https://proxy.atomscan.com/cosmoshub-lcd/cosmos/staking/v1beta1/validators?page.offset=1&pagination.limit=500&status=BOND_STATUS_BONDED"
-	return fetchCosmosSDKNakaCoeff("cosmos", url)
+type cosmosPoolResponse struct {
+	Pool struct {
+		NotBondedTokens string `json:"not_bonded_tokens"`
+		BondedTokens    string `json:"bonded_tokens"`
+	} `json:"pool"`
 }
 
-// fetchCosmosSDKNakaCoeff returns the nakamoto coefficient for the provided cosmos SDK-based chain using the provided url.
-func fetchCosmosSDKNakaCoeff(chainName, url string) (int, error) {
+func FetchCosmosSDKNakaCoeff(chainName, validatorURL, poolURL string) (int, error) {
```

Comment:
add back the comment:
`// fetchCosmosSDKNakaCoeff returns the nakamoto coefficient for the provided cosmos SDK-based chain using the provided url.`

Reply:
Re-added.